### DISCUSSION TITLE
airflow can not access and operation not allowed

### DISCUSSION THREAD

**Discussion by iRecursion** [apache/airflow#25953] [Q&A]
i install airflow 2.3.3 with docker compose on centos 2009 7.9,
when adding files like dag,or logs generating by airflow,
it can not be accessed or operation not allowed!
also can't see in webui,there is a line in $AIRFLOW_HOME/Dockerfile,

`**echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${AIRFLOW_USER_HOME_DIR}:/sbin/nologin" \**`

is there any solution to solve the issue?

`[](https://github.com/iRecursion/asciidoc-module/blob/main/airflow_233compose_access_operation_failure.png)`

**Comment by iRecursion** [comment]
just like https://community.oracle.com/tech/apps-infra/discussion/4491730/docker-image-oraclelinux-8-ls-shows-operation-not-permitted

**Reply by potiuk** [reply]
This is problem with the centos not airflow. You should ask your question there. Or switch to official Airlfow image which is based on Debian. We have an open proposal on how we could support Centos https://github.com/apache/airflow/pull/24825 - but we do not have community based "official" reference image for CentOS. You can take a look at the Dockerfile in #24825 and compare it with yours - maybe you will find a solution there.

**Reply by potiuk** [reply]
Also, if you want to stick with CentOS not our official image, I heartily recommend to read all the documentation there https://airflow.apache.org/docs/docker-stack/index.html if you want to build your own image. It contains a lot of explanation and discussion why and how we build our image, what the entrypoint is doing, explaining the signal propagation and many other things. That might be an invaluable sources for you to learn how to avoid a number of problems when you build your own custom image.

**Reply by potiuk** [reply]
You can also watch my talk about it from Airflow Summit 2020: https://www.youtube.com/watch?v=wDr3Y7q2XoI - it gives a lot of insights on how we approached our image and might give you the feeling what you are up to when you go custom route.

**Reply by iRecursion** [reply]
> This is problem with the centos not airflow. You should ask your question there. Or switch to official Airlfow image which is based on Debian. We have an open proposal on how we could support Centos #24825 - but we do not have community based "official" reference image for CentOS. You can take a look at the Dockerfile in #24825 and compare it with yours - maybe you will find a solution there.

i'm ready to customize the image, thanks for your very prompt and helpful reply!
